normits_demand/models/notem/attraction_models.py

```diff
@@ -299,24 +302,23 @@ def run(self,
                 )
 
             # ## SPLIT PURE ATTRACTIONS BY MODE ## #
-            du.print_w_toggle("Splitting by mode...", verbose=verbose)
+            self._logger.info("Splitting by mode")
             fully_segmented = self._split_by_mode(pure_attractions)
 
             # ## ATTRACTIONS TOTAL CHECK ## #
             if not pure_attractions.sum_is_close(fully_segmented):
-                warnings.warn(
+                msg = (
                     "The attraction totals before and after mode split are not same.\n"
                     "Expected %f\n"
                     "Got %f"
                     % (pure_attractions.sum(), fully_segmented.sum())
-                )
+                )                
+                self._logger.warning(msg)
+                raise warnings.warn(msg)
```

Comment:
This is the wrong calling convention for warnings. You don't raise warnings in this way, just call them with `warnings.warn(msg)` - you'll probably want to update this in all places